core/types/transaction_signing.go

```diff
@@ -102,6 +104,78 @@ type Signer interface {
 	Equal(Signer) bool
 }
 
+type EIP2718Signer struct{ EIP155Signer }
+
+func NewEIP2718Signer(chainId *big.Int) EIP2718Signer {
+	return EIP2718Signer{
+		NewEIP155Signer(chainId),
+	}
+}
+
+// Sender returns the recovered addressed from a transaction's signature.
+// It assumes V does not store the chain id, unless the tx is of legacy type.
+func (s EIP2718Signer) Sender(tx *Transaction) (common.Address, error) {
+	if !tx.Protected() {
+		return HomesteadSigner{}.Sender(tx)
+	}
```

Comment:
Is there an implicit assumption here that non-legacy transactions are protected? (if so: I don't think that assumption should be made)

Reply:
Unless it's clearly specified in the 2718-EIP that new-style transactions are `Protected`

Reply:
I reordered the function so it is clear that there is no assumption as to whether typed txs are protected.